frontend/src/test/java/com/google/sps/InMemoryTripModelStorageTest.java

```diff
@@ -0,0 +1,266 @@
+package com.google.sps;
+import static com.google.common.truth.Truth.assertThat;
+import static com.google.common.truth.Truth8.assertThat;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import java.util.List;
+import com.google.tripmeout.frontend.storage.InMemoryTripModelStorage;
+import com.google.tripmeout.frontend.TripModel;
+import com.google.tripmeout.frontend.error.TripAlreadyExistsException;
+import com.google.tripmeout.frontend.error.TripNotFoundException;
+
+
+public class InMemoryTripModelStorageTest {
+  public static final TripModel LONDON_TRIP1 = TripModel.builder()
+      .setId("a")
+      .setName("London Trip")
+      .setUserId("person 1")
+      .setLocationLat(32.333)
+      .setLocationLong(52.1)
+      .build();
+
+  public static final TripModel LONDON_TRIP2 = TripModel.builder()
+      .setId("b")
+      .setName("London Trip")
+      .setUserId("person 2")
+      .setLocationLat(32.333)
+      .setLocationLong(52.1)
+      .build();
+
+  public static final TripModel LONDON_TRIP1_2022 = TripModel.builder()
+      .setId("c")
+      .setName("London Trip 2022")
+      .setUserId("person 1")
+      .setLocationLat(32.333)
+      .setLocationLong(52.1)
+      .build();
+
+  public static final TripModel TOKYO_TRIP1 = TripModel.builder()
+      .setId("d")
+      .setName("Tokyo Trip")
+      .setUserId("person 1")
+      .setLocationLat(32.333)
+      .setLocationLong(147.1112)
+      .build();
+
+  public static final TripModel LONDON_TRIP_DUPLICATE_ID = TripModel.builder()
+      .setId("a")
+      .setName("London Trip")
+      .setUserId("person 2")
+      .setLocationLat(32.333)
+      .setLocationLong(52.1)
+      .build();
+
+
+  /**
+   * test that an exception is thrown when adding a TripModel that has already 
+   * been added to storage
+   */
+  @Test
+  public void addTrip_tripAlreadyAdded_throwsException() 
```

Comment:
Nit: it seems like the condition you want to test here is that a trip with the _same ID_ already exists. If so, I recommend updating the description to match that; you may even want to change the test case to try and store a _different_ trip model object that happens to have the same key. (I'll leave that choice up to you; doesn't seem super important here.)

Reply:
Just realized that the matching trip ID case is already handled. In that case, you can just remove this test case. In general, you want to test exactly the edge cases you care about in as few ways/places as possible to make refactoring easier if, say, the interface semantics change.